packages/preferences/src/browser/preference-service.spec.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { Container } from 'inversify';
+import * as chai from 'chai';
+import { Emitter } from '@theia/core/lib/common';
+import { PreferenceService, PreferenceServiceImpl } from '@theia/preferences-api/lib/browser/';
+import { FileSystemWatcher, FileSystem, FileChange, } from '@theia/filesystem/lib/common/';
+import { FileSystemWatcherServer } from '@theia/filesystem/lib/common/filesystem-watcher-protocol';
+import { ILogger } from '@theia/core/lib/common/logger';
+import { PreferenceProviderUser, PreferenceProviderWorkspace } from './';
```

Comment:
index.ts should be used to expose API for external clients, using it internally can lead to cycles

Reply:
Good to know thank you!